WordPress/gutenberg
# Use core version of template and template part post types and REST endpoints for WP 5.9, with back compat for 5.8

## Description
When the Gutenberg plugin is active, it's currently using it's own version of the template  and template part post types and REST endpoints.

Usually the practice is to rely on the core version of PHP code when it exists.

`register_post_type` overwrites any previous registration and we don't want it to overwrite the core version of the post types. The fix here is to check whether the post types have already been registered before attempting to register them again. 

This also solves the REST API issue, as that's also defined as part of the post type configuration.

## How has this been tested?
1. Make sure you're running WordPress 5.9
2. Change something related to one of the post types in the Gutenberg codebase (i.e. one of the labels)
3. Load the relevant part of the UI and notice the change is not displayed because this code isn't active.

## Types of changes

Bug fix (non-breaking change which fixes an issue)



## Checklist:
